spec/integration/client/exit_code_spec.rb

```diff
@@ -0,0 +1,264 @@
+        :cwd => chef_dir,
+        :returns => [exit_code])
+    end
+
+    context "has a cookbook" do
+
+      context "with a library" do
+
+        context "which cannot be loaded" do
+          before do
+            file "cookbooks/x/recipes/default.rb", ""
+            file "cookbooks/x/libraries/error.rb", "require 'does/not/exist'"
+          end
+
+          it "exits with GENERAL_FAILURE, 1" do
+            setup_client_rb
+            run_chef_client_and_expect_exit_code 1
+          end
+        end
+
+      end
+
+      context "with an audit recipe" do
+
+        context "which fails" do
+          before do
+            file "cookbooks/x/recipes/default.rb", <<-RECIPE
+control_group "control group without top level control" do
+  it "should fail" do
+    expect(4 - 4).to eq(1)
+  end
+end
+RECIPE
+          end
+
+          it "exits with AUDIT_MODE_FAILURE, 42" do
+            setup_client_rb_with_audit_mode
+            run_chef_client_and_expect_exit_code 42
+          end
+        end
+
+      end
+
+      context "with a recipe" do
+
+        context "which throws an error" do
+          before { file "cookbooks/x/recipes/default.rb", "raise 'BOOM'" }
+
+          it "exits with GENERAL_FAILURE, 1" do
+            setup_client_rb
+            run_chef_client_and_expect_exit_code 1
+          end
+        end
+
+        context "with a recipe which calls Chef::Application.fatal with a non-RFC exit code" do
+          before { file "cookbooks/x/recipes/default.rb", "Chef::Application.fatal!('BOOM', 123)" }
+
+          it "exits with the GENERAL_FAILURE exit code, 1" do
+            setup_client_rb
+            run_chef_client_and_expect_exit_code 1
+          end
+        end
+
+        context "with a recipe which calls Chef::Application.exit with a non-RFC exit code" do
+          before { file "cookbooks/x/recipes/default.rb", "Chef::Application.exit!('BOOM', 231)" }
+
+          it "exits with the GENERAL_FAILURE exit code, 1" do
+            setup_client_rb
+            run_chef_client_and_expect_exit_code 1
+          end
+        end
+
+        context "when a reboot exception is raised (like from the reboot resource)" do
+          before do
+            file "cookbooks/x/recipes/default.rb", <<EOM
+raise Chef::Exceptions::Reboot.new
+EOM
+          end
+
+          it "exits with REBOOT_SCHEDULED, 35" do
+            setup_client_rb
+            run_chef_client_and_expect_exit_code 35
+          end
+        end
+        
+        context "when an attempt to reboot fails (like from the reboot resource)" do
+          before do
+            file "cookbooks/x/recipes/default.rb", <<EOM
+raise Chef::Exceptions::RebootFailed.new
+EOM
+          end
+
+          it "exits with REBOOT_FAILED, 41" do
+            setup_client_rb
+            run_chef_client_and_expect_exit_code 41
+          end
+        end
+
+      end
+
+    end
+
+  end
+
```

Comment:
Can you delete the extra lines between end statements and between context statements in this file?